src/__tests__/enum.test.ts

```diff
@@ -84,6 +91,21 @@ test('should create a schema', () => {
             },
           ],
         },
+        {
+          "customProp": "val",
+          "name": "enum_field_with_customProp",
+          "type": {
+            "customProp": "val",
```

Comment:
Please consider if the custom property should be duplicated here on top and nested level. I though since the name is duplicated the custom prop also should be, but either way is fine. This also applies to a few other types. (Any that have the nested/duplicate name.